version: '3.1'

services:
  django:
    platform: linux/amd64
    build:
      context: .
      dockerfile: ./compose/local/django/Dockerfile
    depends_on:
      - postgres
    volumes:
      - .:/app:z
    init: true
    environment:
      DATABASE_URL: "postgres://postgres:admin@postgres:5432/ds_judgements_public_ui"
      DJANGO_SETTINGS_MODULE: "config.settings.local"
      SECRET_KEY: local_dev_secret_key
      SECURE_SSL_REDIRECT: "false"
    env_file:
      - ./.env
    ports:
      - "3000:3000"
    # do nothing forever - exec commands elsewhere
    command: tail -f /dev/null

  postgres:
    image: postgres:12.3
    ports:
     - "5432:5432"
    environment:
      POSTGRES_DB: ds_judgements_public_ui
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: admin
      POSTGRES_HOST: postgres
      POSTGRES_PORT: 5432
      PGDATABASE: ds_judgements_public_ui
      PGUSER: postgres
      PGPASSWORD: admin
      POSTGRES_HOST_AUTH_METHOD: trust
    volumes:
      - ./database_backups:/backups:z

  marklogic:
    container_name: marklogic
    image: store/marklogicdb/marklogic-server:10.0-8.1-centos-1.0.0-ea2
    ports:
      - "8000:8000"
      - "8001:8001"
      - "8002:8002"
      - "8011:8011"
    environment:
      MARKLOGIC_ADMIN_USERNAME: admin
      MARKLOGIC_ADMIN_PASSWORD: admin
      MARKLOGIC_INIT: "true"
    volumes:
      - ./docker/db/data/:/var/opt/MarkLogic/
      - ./docker/db/backup:/var/opt/backup
